src/contexts/ScreenSize.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { BreakPoint } from '@aragon/ui'
+
+const SMALL = 'small'
+const MEDIUM = 'medium'
+const LARGE = 'large'
+
+const { Provider, Consumer } = React.createContext()
+
+const ScreenSizeProvider = ({ children }) => (
+  <React.Fragment>
+    <BreakPoint to="medium">
+      <Provider value={{ screenSize: SMALL }}>{children}</Provider>
+    </BreakPoint>
+    <BreakPoint from="medium" to="large">
+      <Provider value={{ screenSize: MEDIUM }}>{children}</Provider>
+    </BreakPoint>
+    <BreakPoint from="large">
+      <Provider value={{ screenSize: LARGE }}>{children}</Provider>
+    </BreakPoint>
+  </React.Fragment>
+)
```

Comment:
The problem of the `BreakPoint` approach (or [`react-media`](https://github.com/ReactTraining/react-media) more exactly) is that it unmounts / mounts the entire subtree, which triggers the apps menu and the current app to reload, plus other things we don’t see.

A solution would be to have multiple queries passed to `Media`, passing their result in a single render prop… but I don’t see how to solve this without modifying `Media`. 🤔

I see two solutions:

- Do what `Media` is doing, by setting the breakpoints and listening to them using `window.matchMedia()`. It’s probably the simplest?
- Listen to `window.onResize` and get `window.innerWidth` (à la [`GetWindowSize`](https://github.com/aragon/aragon-apps/blob/master/apps/finance/app/src/components/GetWindowSize.js)). It would need to be throttled, and to only trigger updates when a breakpoint has been passed. The advantage it could give would be to get more frequent updates on the viewport width when needed, which could help in some cases. But until we need it, we should probably go for the `window.matchMedia()` approach.

What do you think?

Reply:
>  it unmounts / mounts the entire subtree

Yes, I saw that. Consider the use case: a user resizing their window. I think this is something that we will fix when we implement [this change](https://gist.github.com/bpierre/cd329530c5eaac09bb25a4f79d534de9#file-menupanel-jsx-L324-L330) you suggested.

@bpierre  do you see this as a blocker to continue with the current implementation or can I push it to the backlog and work on it later?

Reply:
> @bpierre do you see this as a blocker to continue with the current implementation or can I push it to the backlog and work on it later?

No I think we can have it like this for now 👍